### ISSUE TITLE
Black bar issue on 1200p desktop resolution

### ISSUE THREAD

**Issue by papahuge** [clsid2/mpc-hc#2223]
Whenever I make the video fullscreen in MPCHC64 latest version on my 1200p native monitor, the bottom inch gets a black bar. Cutting into the content of the videos. It even happens when I stretch the video box to full screen dimensions without going fullscreen mode. This has been happening for a long time, so i'm not sure if it has anything to do with recent update. Please take a look... [https://youtu.be/siGg8xzAy5I](https://youtu.be/siGg8xzAy5I)

This effect doesn't happen on my secondary 1080p monitor, which leads me to believe it's a 1200p resolution issue.

**Comment by clsid2** [comment]
Use default settings:
Menu > View > Video Frame > Touch Window from Inside

**Comment by papahuge** [comment]
@clsid2 

> Use default settings: Menu > View > Video Frame > Touch Window from Inside

Sir did you even watch the video I made? Because there is a point in the video where I go through many of the 'video frame' settings. I've tried that setting and all the other ones... It doesn't fix the problem. I believe the issue is in the code somewhere, but I wouldn't know where to start looking. Maybe some kind of masking situation for 1200p resolutions.

Addendum: 
Perhaps this could be that in windows settings, my (1) display is the 1080p one (actually my second monitor on the right). And the secondary (2) display is actually being used as my main 1200p display (on the left). Perhaps the program thinks that the entire system is limited to 1080p and is trying to crop the rendering to 1080 on a 1200p display. So it doesn't render that extra 120 pixels.

![display](https://github.com/clsid2/mpc-hc/assets/73111061/77ad5cff-1daf-4b14-8daa-01afb1da0291)

**Comment by clsid2** [comment]
HKEY_CURRENT_USER\Software\MPC-HC\MPC-HC\Settings
Set "DesktopSizeBackBuffer" to 1.

Does that help?

**Comment by papahuge** [comment]
> Set "DesktopSizeBackBuffer" to 1.

Wow yeah that fixed it right up! Thanks! Now full screen works perfectly on both monitors. Curious as to why this was happening, was it because of the reverse order of monitors? Or something else entirely?

Moar:
Okay now the setting gets reset multiple times when I close the player and regedit, after changing it 4-5 times it finally sticks. But I seem to have figured out why this is happening or at least from practical standpoint...

It happens because the player is closed or exited on the 1080p monitor. Then when it loads back up it loads on the 1080p monitor in same spot, but when I move it to the 1200p display the black bar shows up. Now... when I close/exit the player on the 1200p monitor upon reload it will function properly. Whether the registry setting is set to 1 or 0. 😉

**Comment by clsid2** [comment]
Something must be going wrong with the monitor detection, because it normally already makes sure the backbuffer is big enough for the resolution of the largest monitor.
Is the second monitor enabled when you load the player? Here the second monitor is detected even when turned off, but it may depend on the type of connection.

I need to make a test build for you to figure out what goes wrong.

**Comment by clsid2** [comment]
Here is test build:
removed

It should show some messageboxes with info about the desktop size, the detected monitors, and backbuffer.

**Comment by papahuge** [comment]
here are the results...
Desktop size 3840x1200
Monitor 1 = 1920x1200 (weird since display1 is 1080)
Monitor 2 = 1920x1200
Screen 1920x1080 | Backbuffer 1920x1200

this was opened with the program on the right 1080p monitor.

But I think I finally found what's causing it... it's the Shaders!! I've narrowed it down to post-resize pixel shaders. I should have mentioned that I was using some of the built in shaders to enhance the old 90s trek videos. But basically it doesn't matter what shader is being used, as long as there is at least one active shader in the 'post-resize pixel shaders' box, this issue happens. After I remove the post-resize shaders the issue goes away entirely. 

These are the shaders I'm using...
![shaders](https://github.com/clsid2/mpc-hc/assets/73111061/c2bb14f5-22c6-4a96-b0f5-2096f8e496e0)
